app/Http/Controllers/ScoresController.php

```diff
@@ -42,6 +44,10 @@ private static function parseIdOrFail(string $id): int
 
     public function download($rulesetOrSoloId, $id = null)
     {
+        if(from_app_url() && \Auth::user() === null) {
```

Comment:
move and use the currentUser variable, and the api/web check function is the other one (it's right in the name)

Reply:
But isn't from_app_url() checking if called via the website? That's what we wanna check for, no? Like, the osu.ppy.sh/scores/id/download url. The API endpoint /api/v2/scores/id/download works as expected and fails if not authorized

Reply:
I hope I finally understood how things work, thanks for being patient with me and giving me the chance to learn this 😅